common/isolationgroup/defaultisolationgroupstate/state.go

```diff
@@ -163,22 +169,45 @@ func (z *defaultIsolationGroupStateHandler) get(ctx context.Context, domain stri
 	return ig, nil
 }
 
+func (z *defaultIsolationGroupStateHandler) createAvailableisolationGroupMetricsScope(domainID string, availableIsolationGroups []string) metrics.Scope {
```

Comment:
I think adding the tasklist ID being passed in here would be simple enough and provide that additional dimension, should we need it (which think we do)